packages/shared/src/components/post/PostSummaryButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { ReactElement } from 'react';
+import { useSummaryActivation } from '../../hooks/experiments';
+import { Loader } from '../Loader';
+import { MagicIcon } from '../icons';
+import { ButtonIconPosition, ButtonVariant } from '../buttons/common';
+import { Button, ButtonColor } from '../buttons/Button';
+
+interface PostSummaryButtonProps {
+  summary: string;
```

Comment:
type wise summary can be undefined, I know TS is not complaining now but in strict TS it would (same for hook).